workers/unity/Packages/io.improbable.gdk.tools/.CodeGenTemplate/CodeGen/EntryPoint.cs

```diff
@@ -43,27 +49,88 @@ public CodeGenerator(CodeGeneratorOptions options, IFileSystem fileSystem)
         {
             this.options = options;
             this.fileSystem = fileSystem;
+
+            ConfigureLogger();
+            logger = LogManager.GetCurrentClassLogger();
+        }
+
+        private static void ConfigureLogger()
+        {
+            var codeGeneratorOptions =  CodeGeneratorOptions.Instance;
+
+            var config = new NLog.Config.LoggingConfiguration();
+
+            var jsonLayout = new JsonLayout
+            {
+                Attributes =
+                {
+                    new JsonAttribute("time", "${longdate}"),
+                    new JsonAttribute("level", "${level:uppercase=true}"),
+                    new JsonAttribute("logger", "${logger}"),
+                    new JsonAttribute("message", "${message}"),
+                    new JsonAttribute("exception", "${exception:format=ToString}")
+                }
+            };
+
+            var minimumLogLevel = codeGeneratorOptions.DisableVerboseLogging ? LogLevel.Info : LogLevel.Trace;
+
+            if (codeGeneratorOptions.EnableLoggingToConsole)
+            {
+                var consoleTarget = new ConsoleTarget("consoleTarget")
+                {
+                    Layout = jsonLayout
+                };
+                config.AddTarget(consoleTarget);
+                config.AddRule(minimumLogLevel, LogLevel.Fatal, consoleTarget);
+            }
+
+            var fileTarget = new FileTarget("fileTarget")
+            {
+                FileName = codeGeneratorOptions.AbsoluteLogPath,
+                Layout = jsonLayout,
```

Comment:
The log file should be human readable imo (i.e. - not JSON). The Editor should read from `stdout` (since it has exclusive access to that) and the logfile should be for the user